torch/distributed/_shard/checkpoint/storage.py

```diff
@@ -195,3 +186,20 @@ def read_metadata(self) -> Metadata:
 
         """
         pass
+
+class CheckpointException(BaseException):
```

Comment:
nit: Probably might be better to put this in a separate file since this file seems to be just the interfaces for StorageReader/Writer?